src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerDriver.GroupedAnalyzerActions.cs

```diff
@@ -74,7 +86,25 @@ IGroupedAnalyzerActions IGroupedAnalyzerActions.Append(IGroupedAnalyzerActions i
 
                 var newGroupedActions = GroupedActionsByAnalyzer.AddRange(groupedAnalyzerActions.GroupedActionsByAnalyzer);
                 var newAnalyzerActions = AnalyzerActions.Append(groupedAnalyzerActions.AnalyzerActions);
-                return new GroupedAnalyzerActions(newGroupedActions, newAnalyzerActions);
+                var analyzersByKind = CreateAnalyzersByKind(newGroupedActions);
+                return new GroupedAnalyzerActions(newGroupedActions, analyzersByKind, newAnalyzerActions);
+            }
+
+            private static ImmutableSegmentedDictionary<TLanguageKindEnum, ImmutableArray<DiagnosticAnalyzer>> CreateAnalyzersByKind(ImmutableArray<(DiagnosticAnalyzer, GroupedAnalyzerActionsForAnalyzer)> groupedActionsAndAnalyzers)
+            {
+                var analyzersByKind = PooledDictionary<TLanguageKindEnum, ArrayBuilder<DiagnosticAnalyzer>>.GetInstance();
+                foreach (var (analyzer, groupedActionsForAnalyzer) in groupedActionsAndAnalyzers)
+                {
+                    foreach (var (kind, _) in groupedActionsForAnalyzer.NodeActionsByAnalyzerAndKind)
+                    {
+                        analyzersByKind.AddPooled(kind, analyzer);
+                    }
+                }
+
+                var result = analyzersByKind.ToImmutableSegmentedDictionaryAndFree();
```

Comment:
consider making ToImmutableSegmentedDictionaryAndFree an extension on PooledDict, and have it free the pooled dict as well (see my pr).  there's only one other case you need to fix up.  it simplifies things nicely.

Reply:
See https://github.com/dotnet/roslyn/pull/76925/commits/29adf56957c26aef88f2c020014b04542965e5d1

Reply:
done!